From 04a06a459a2e6123a884be08f200568f947b4cc8 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 9 Jan 2013 14:42:11 +0200
Subject: [PATCH 045/177] lcd3-cape: Convert to using the proper touchscreen
 driver.

Now that the touchscreen driver has DT bindings switch to using it.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 firmware/capes/cape-bone-lcd3-00A0.dts | 33 +++++++++++++++------------------
 1 file changed, 15 insertions(+), 18 deletions(-)

diff --git a/firmware/capes/cape-bone-lcd3-00A0.dts b/firmware/capes/cape-bone-lcd3-00A0.dts
index 83a6d88..2d806c8 100644
--- a/firmware/capes/cape-bone-lcd3-00A0.dts
+++ b/firmware/capes/cape-bone-lcd3-00A0.dts
@@ -87,13 +87,23 @@
 			#size-cells = <1>;
 
 			tscadc {
-				compatible = "ti-tscadc-dt";
+				compatible = "ti,ti-tscadc";
+				reg = <0x44e0d000 0x1000>;
 
-				tsc-wires = <4>;
-				tsc-x-plate-resistance = <200>;
-				tsc-steps = <6>;
+				interrupt-parent = <&intc>;
+				interrupts = <16>;
+				ti,hwmods = "adc_tsc";
+
+				tsc {
+					wires = <4>;
+					x-plate-resistance = <200>;
+					steps-to-configure = <5>;
+					wire-config = <0x00 0x11 0x22 0x33>;
+				};
 
-				adc-channels = <4>;
+				adc {
+					adc-channels = <4>;
+				};
 			};
 
 			gpio-leds-cape-lcd3 {
@@ -166,19 +176,6 @@
 				};
 			};
 
-//			lcd3-cape {
-//				compatible = "da8xx-dt";
-//
-//				pinctrl-names = "default";
-//				pinctrl-0 = <&bone_lcd3_cape_lcd_pins>;
-//
-//				ti,hwmods = "lcdc";
-//
-//				disp-pll = <16000000>;
-//				panel-type = "CDTech_S035Q01";
-//
-//			};
-
 			/* Settings for CDTech_S035Q01 / LCD3 cape: */
 			panel {
 				compatible = "lcdc,panel";
-- 
1.8.1.4

